cognite_toolkit/_cdf_tk/resource_classes/function_schedule.py

```diff
@@ -0,0 +1,51 @@
+from croniter import croniter
+from pydantic import Field, field_validator
+
+from .base import ToolkitResource
+
+
+class FunctionAuthenticationField(ToolkitResource):
```

Comment:
We have the same for workflow trigger (and soon transformations), so lets just call this something like this
```suggestion
class AuthenticationClientIdSecret(ToolkitResource):
```